src/compile/scale/domain.ts

```diff
@@ -162,3 +162,67 @@ function _useRawDomain (scale: Scale, model: Model, channel: Channel) {
       (fieldDef.type === 'temporal' && util.contains([ScaleType.TIME, ScaleType.UTC], scale.type))
     );
 }
+
+
+/**
+ * Convert the domain to an array of data refs or an array of values. Also, throw
+ * away sorting information since we always sort the domain when we union two domains.
+ */
+function normalizeDomain(domain: VgDomain): (any[] | VgDataRef)[] {
+  if (util.isArray(domain)) {
+    return [domain];
+  } if (isDataRefDomain(domain)) {
+    delete domain.sort;
```

Comment:
This still doesn't throw warning, I just added #1895 for this so we don't have to deal with it now.